src/runtime_src/core/common/xclbin_parser.h

```diff
@@ -95,16 +95,44 @@ struct softkernel_object
   char *sk_buf;
 };
 
+// struct aie_cdo_obj - wrapper for an AIE CDO group object
+//
+// @cdo_name: CDO group name
+// @cdo_type: CDO group type
+// @pdi_id: ID of this CDO group in PDI
+// @kernel_id: Kernel ID associated with this CDO group
+struct aie_cdo_group_obj
+{
+  std::string cdo_name;
+  uint8_t cdo_type;
+  uint64_t pdi_id;
+  uint64_t kernel_id;
+};
+
+// struct aie_pdi_obj - wrapper for an AIE PDI object
+//
+// @uuid: PDI UUID
+// @cdo_groups: Array of CDO groups
+// @pdi: PDI blob
+struct aie_pdi_obj
+{
+  xuid_t uuid;
```

Comment:
I wonder who the consumer is, but given that this is C++ code I think it should be xrt::uuid instead?
xrt::uuid can be constructed from an xuid_t and is of course doing the same copy you are explicitly doing.

Reply:
The consumer will be user space c++ code, e.g. hw_emu, windows user space code. I updated it use xrt::uuid. Thanks.